src/core/reactor.cc

```diff
@@ -3648,14 +3648,11 @@ void smp::create_thread(std::function<void ()> thread_loop) {
 // in the whole program and that after it is invoked the default handler is restored.
 template<int Signal, void(*Func)()>
 void install_oneshot_signal_handler() {
-    static bool handled = false;
     static util::spinlock lock;
 
     struct sigaction sa;
     sa.sa_sigaction = [](int sig, siginfo_t *info, void *p) {
-        std::lock_guard<util::spinlock> g(lock);
-        if (!handled) {
-            handled = true;
+        if (lock.try_lock()) {
```

Comment:
Wait, this is more tricky. We also have a wasmtime signal handler. We need to understand the clash between the seastar handler and the wasmtime handler.

Without that, I don't think we can have nested calls.

Reply:
I'm confused, isn't it in scylla?
Don't we use the oneshot signal handler only in the following cases:
```
src/core/reactor.cc:3891:    install_oneshot_signal_handler<SIGSEGV, sigsegv_action>();
src/core/reactor.cc:3895:    install_oneshot_signal_handler<SIGABRT, sigabrt_action>();
```

Reply:
Cc @psarna what vectors does wasm runtime use?

Reply:
@bhalevy just did a short investigation, here's the current source code from wasmtime which registers signal handlers for UNIX-like systems:
https://github.com/bytecodealliance/wasmtime/blob/601e8f30949f34d2ebec6d93c60531fb7e8c7610/crates/runtime/src/traphandlers/unix.rs#L43-L60
```rust
    // Allow handling OOB with signals on all architectures
    register(&mut PREV_SIGSEGV, libc::SIGSEGV);

    // Handle `unreachable` instructions which execute `ud2` right now
    register(&mut PREV_SIGILL, libc::SIGILL);

    // x86 and s390x use SIGFPE to report division by zero
    if cfg!(target_arch = "x86") || cfg!(target_arch = "x86_64") || cfg!(target_arch = "s390x") {
        register(&mut PREV_SIGFPE, libc::SIGFPE);
    }

    // Sometimes we need to handle SIGBUS too:
    // - On ARM, handle Unaligned Accesses.
    // - On Darwin, guard page accesses are raised as SIGBUS.
    if cfg!(target_arch = "arm") || cfg!(target_os = "macos") || cfg!(target_os = "freebsd") {
        register(&mut PREV_SIGBUS, libc::SIGBUS);
    }
```

I also preventively looked for `SIGUSR1` on their github and it's indeed being discussed right now as one of the possible foundations for more exact fuel metering: https://github.com/bytecodealliance/wasmtime/issues/4109

Reply:
Ugh tell them to stop